pkg/backuphelpers/backupvars.go

```diff
@@ -0,0 +1,98 @@
+package backuphelpers
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+	"sync"
+
+	backupv1alpha1 "github.com/openshift/api/config/v1alpha1"
+	prune "github.com/openshift/cluster-etcd-operator/pkg/cmd/prune-backups"
+)
+
+type Enqueueable interface {
+	Enqueue()
+}
+
+type BackupVar interface {
+	AddListener(listener Enqueueable)
+	SetBackupSpec(spec *backupv1alpha1.EtcdBackupSpec)
+	ArgString() string
+}
+
+type BackupConfig struct {
+	enabled   bool
+	spec      *backupv1alpha1.EtcdBackupSpec
+	listeners []Enqueueable
+	mux       sync.Mutex
+}
+
+func NewDisabledBackupConfig() *BackupConfig {
+	return &BackupConfig{
+		enabled: false,
+		mux:     sync.Mutex{},
+	}
+}
+
+func (b *BackupConfig) SetBackupSpec(spec *backupv1alpha1.EtcdBackupSpec) {
+	b.mux.Lock()
+	defer b.mux.Unlock()
+
+	if reflect.DeepEqual(b.spec, spec) {
+		return
+	}
+
+	b.enabled = spec != nil
+
+	if spec == nil {
+		b.spec = nil
+	} else {
+		specClone := *spec
+		b.spec = &specClone
```

Comment:
I leave this for another PR, but you could also simply do a 
```suggestion
		b.spec = spec.DeepCopy()
```

Reply:
done 👍🏽